### PR TITLE
Try to reuse extension_is_relevant() logic

### PR DESCRIPTION
**Pull request by kaduk**
Noticed this while working on something else, so don't have enough time to focus on getting the details right.
(This currently isn't right; test_sslmessages complains at least.)

Or maybe @mattcaswell remembers a reason why it won't work.

[skip ci] (to be removed when coming out of WIP)

### CODE REVIEW

**Comment by mattcaswell** [comment]
Ping @kaduk - what is happening with this PR?

**Comment by kaduk** [comment]
@mattcaswell I haven't gotten to look at this since basically when I first put it in.  I'm curious if you think it's worth attempting to consolidate this stuff; if not, we should just close this.

**Comment by mattcaswell** [comment]
I'm in favour of code de-duplication as a principle. It looks like the code has changed a little since you put this in though. Also the conditions in extension_is_relevant() are similar to those used here, but not quite the same (in the latest version of the code). I'm not sure how easy it will be to consolidate and how much benefit is obtained. Your call though.

**Comment by kaduk** [comment]
Taking out of WIP, having spent the time to figure out the subtleties of the differences between the two expressions.

**Comment by mattcaswell** [comment]
@kaduk - needs a rebase

**Comment by kaduk** [comment]
Travis failure is not relevant (apt-get issue).
I think @mattcaswell is still the best reviewer for this even though I am trying to not overload him with too many review requests.

**Comment by levitte** [comment]
Ping @mattcaswell

File: ssl/statem/extensions.c [hunk 1]
```diff
@@ -498,11 +498,20 @@ int extension_is_relevant(SSL *s, unsigned int extctx, unsigned int thisctx)
                 && (extctx & SSL_EXT_TLS_IMPLEMENTATION_ONLY) != 0)
             || (s->version == SSL3_VERSION
                     && (extctx & SSL_EXT_SSL3_ALLOWED) == 0)
+            /*
+             * Note that SSL_IS_TLS() means "TLS 1.3 has been negotiated",
```
Comment (mattcaswell) [inline ssl/statem/extensions.c]: Nit: You mean SSL_IS_TLS13() right?

**Comment by kaduk** [comment]
Pushed with nit fixed; closing.  Thanks for the review!

**Pull request closed without merging**